Training3/src/task/key-logger/src/component/modal.tsx

```diff
@@ -0,0 +1,50 @@
+import React, {useEffect, useRef, useState} from 'react';
+
+type ModalProps = {
+    isOpen: boolean,
+    handleClose: () => void,
+    onKeyPress: (text: string) => void,
+}
+
+const Modal = ({isOpen, handleClose, onKeyPress}: ModalProps) => {
+    const ref = useRef<HTMLTextAreaElement>(null)
+    const [isRendered, setIsRendered] = useState<boolean>(false)
+    let display = isOpen ? 'block' : 'none';
+
+    useEffect(() => {
+        if (isOpen) {
+            setIsRendered(true)
+        }
+    }, [isOpen]);
+
+    useEffect(() => {
+        const handleTxtAreaChange = (e: KeyboardEvent) => {
+            onKeyPress(e.key);
+        }
+
+        if (ref.current) {
+            ref.current.addEventListener("keydown", handleTxtAreaChange)
+        }
+
+        return () => {
+            if (ref.current) {
```

Comment:
duplicate check logic ref.current, should we move return to first if?

if (ref.current) {
            ref.current.addEventListener("keydown", handleTxtAreaChange)
            return () => {
              ref.current.removeEventListener("keydown", handleTxtAreaChange)
            }
        }